lib/rubocop/cop/rspec/factory_bot/factory_name_style.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Checks for name style for argument of FactoryBot::Syntax::Methods.
+        #
+        # @example EnforcedStyle: symbol (default)
+        #   # bad
+        #   create('user')
+        #   build "user", username: "NAME"
+        #
+        #   # good
+        #   create(:user)
+        #   build :user, username: "NAME"
+        #
+        # @example EnforcedStyle: string
+        #   # bad
+        #   create(:user)
+        #   build :user, username: "NAME"
+        #
+        #   # good
+        #   create('user')
+        #   build "user", username: "NAME"
+        #
+        class FactoryNameStyle < Base
+          extend AutoCorrector
+          include ConfigurableEnforcedStyle
+          include RuboCop::RSpec::FactoryBot::Language
+
+          MSG = 'Use %<prefer>s to refer to a factory.'
+          FACTORY_CALLS = RuboCop::RSpec::FactoryBot::Language::METHODS
+          RESTRICT_ON_SEND = FACTORY_CALLS
+
+          # @!method factory_call(node)
+          def_node_matcher :factory_call, <<-PATTERN
+          (send
+            {#factory_bot? nil?} %FACTORY_CALLS
+            ${str sym} ...
+          )
+          PATTERN
+
+          def on_send(node)
+            factory_call(node) do |name|
+              if offnse_for_symbol_style?(name)
+                register_offense(name, name.value.to_sym.inspect)
+              elsif offnse_for_string_style?(name)
+                register_offense(name, name.value.to_s.inspect)
+              end
+            end
+          end
+
+          private
+
+          def offnse_for_symbol_style?(name)
+            name.str_type? && style == :symbol
```

Comment:
I'm thinking about non-str, non-symbol definitions like `create("user_#{type}"). Those would be ignored, right?

Reply:
Yes, `create("user_#{type}")` is assumed to be ignored.  For clarity, we added test code.

Reply:
I could only find GitLab doing that:
```
gitlabhq/spec/models/plan_spec.rb
11:        let(:plan) { build("#{plan}_plan".to_sym) }

gitlabhq/spec/models/project_spec.rb
666:      let!(:package) { create("#{package_type}_package", project: project) }
7259:        let!(:package) { create("#{package_type}_package", project: project) }

gitlabhq/spec/features/groups/empty_states_spec.rb
36:            create("closed_#{issuable}", project_relation => project)

...
```

To my personal taste, this could be achieved with a trait:
```ruby
create(issuable, :closed, project_relation => project)
```
instead of a nested factory.
But this is where static analysis would suffer, as `issuable` can hold anything.

Reply:
Thank you very much. If we try to address this, do we need to find the definition on the trait side and suggest that it be rewritten?

Reply:
That sounds like an overkill.
We could probably still raise an offence with a distinct message to use traits, but skip autocorrection.
Or this can be another cop.